manifest: Updated Matter repository to v1.4.2 version (#23080)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Matter fork
+++++++++++

The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, the ``v1.4.2.0`` tag.

The following list summarizes the most important changes inherited from the upstream Matter:

* Updated:

>   * Fixed incorrect memory releases and unhandled exceptions
>   * Group sessions are managed more robustly
>   * The device commissioning process was optimized

Reviewer: ```suggestion
  * Fixed incorrect memory releases and unhandled exceptions.
  * Improved robustness in group session management.
  * Optimized the device commissioning process.
```

Author: Applied,thanks!
